resources/agent.rb

```diff
@@ -35,32 +35,66 @@
                                    "backend-url": ['ws://127.0.0.1:8081'],
                                  }
 action :install do
-  packagecloud_repo new_resource.repo do
-    type value_for_platform_family(
-      %w(rhel fedora amazon) => 'rpm',
-      'default' => 'deb'
-    )
-  end
+  # Linux installation - source package
+  if node['platform'] != 'windows'
+    packagecloud_repo new_resource.repo do
+      type value_for_platform_family(
+        %w(rhel fedora amazon) => 'rpm',
+        'default' => 'deb'
+      )
+    end
 
-  package 'sensu-go-agent' do
-    if latest_version?(new_resource.version)
-      action :upgrade
-    else
-      version new_resource.version
-      action :install
+    # Installs/upgrades sensu-go-agent package
+    package 'sensu-go-agent' do
+      if latest_version?(new_resource.version)
+        action :upgrade
+      else
+        version new_resource.version
+        action :install
+      end
     end
-  end
 
-  # render template at /etc/sensu/agent.yml
-  file ::File.join(new_resource.config_home, 'agent.yml') do
-    content(stringify_keys(new_resource.config))
+    # render template at /etc/sensu/agent.yml for linux
+    file ::File.join(new_resource.config_home, 'agent.yml') do
+      content(stringify_keys(new_resource.config))
+    end
+
+    # Enable and start the sensu-agent service
+    service 'sensu-agent' do
+      if node['platform'] == 'ubuntu' && node['platform_version'].to_f == 14.04
+        provider Chef::Provider::Service::Init
+        action :start
+      else
+        action [:enable, :start]
+      end
+    end
   end
 
-  service 'sensu-agent' do
-    if node['platform'] == 'ubuntu' && node['platform_version'].to_f == 14.04
-      provider Chef::Provider::Service::Init
-      action :start
-    else
+  # Installs msi for Sensu-Go
+  if node['platform'] == 'windows'
+    windows_package 'sensu-go-agent' do
+      source 'https://s3-us-west-2.amazonaws.com/sensu.io/sensu-go/5.10.0/sensu-go-agent_5.10.0.4171_en-US.x64.msi'
+      installer_type :custom
+      version '5.10.0.4171'
+    end
+
+    # Adds install directory to path
+    windows_path 'c:\\Program Files\\sensu\\sensu-agent\\bin'
```

Comment:
don't know too much about windows but should we be using a param or a helper function for this? I know I have installed application in drives and locations other than `c:\Program Files*`

Reply:
This could only be changed for the default provided by the MSI by using `options` on `windows_package` set to `INSTALLDIR=""`. 

That can be implemented, but not many other installers support this option from what I've seen and seems more of an enhancement that required.

Reply:
gotcha makes sense, we can leave it for now and if someone needs to we can enhance it.